mon-pix/app/components/challenge-item-qcm.js

```diff
@@ -1,26 +1,25 @@
 import { action } from '@ember/object';
 import ChallengeItemGeneric from './challenge-item-generic';
-import classic from 'ember-classic-decorator';
+import _ from 'lodash';
+
+export default class ChallengeItemQcm extends ChallengeItemGeneric {
+  answersValue= [];
 
-@classic
-class ChallengeItemQcm extends ChallengeItemGeneric {
   _hasError() {
     return this._getAnswerValue().length < 1;
   }
 
-  // FIXME refactor that
   _getAnswerValue() {
-    return this.$('input[type=checkbox][id^=checkbox_]:checked').map(function() {return this.name; }).get().join(',');
+    return this.answersValue.join(',');
   }
 
   _getErrorMessage() {
     return 'Pour valider, sélectionner au moins une réponse. Sinon, passer.';
   }
 
   @action
-  answerChanged() {
+  answerChanged(checkboxName) {
+    _.indexOf(this.answersValue, checkboxName) === -1 ? this.answersValue.push(checkboxName) : _.remove(this.answersValue, (value) => value === checkboxName);
```

Comment:
Ça pourrait être un simple `if` non ?
```suggestion
    if(_.indexOf(this.answersValue, checkboxName) === -1) {
      this.answersValue.push(checkboxName);
    } else {
      _.pull(this.answersValue, checkboxName);
    }
```

Peut-être qu'un objet `Set` serait utile ici 🤔

Reply:
Je fais une nouvelle proposition avec le Set